apps/manager_app/onboarding/onboarding.h

```diff
@@ -47,4 +56,33 @@ void onboarding_set_static_screen(void);
  */
 const flow_step_t *onboarding_get_step(void);
 
+/**
+ * @brief This API returns the last completed step of the onboarding flow
+ *
+ * @return onboarding_steps_e Enum depicting last recorded step
+ */
+onboarding_steps_e onboarding_get_last_step(void);
+
+/**
+ * @brief This API updates the progress of the onboarding flow. An important
+ * thing to note here is that it only sets the step if the last_step is not
+ * ONBOARDING_COMPLETE (depicting onboarding is already complete) and the
+ * next_step is actually a new step of the onboarding flow (revocation is not
+ * allowed).
```

Comment:
please add the comment around our intention to entertain special cases of step-updates

Reply:
added